### ISSUE TITLE
send-as-attachment@rcalixte: feature request: add filename without extension as subject

### ISSUE THREAD

**Issue by DaddyAG** [linuxmint/cinnamon-spices-actions#568]
### Action name and maintainer

send-as-attachment@rcalixte @rcalixte

### What would you like to see?

Hello,

In windows, when you send attachment via right click on file and "send as attachment", e-mail subject is filled with attachment filename. If you are sending clearly named files, this is very beneficial since it speeds up email writing.

https://linux.die.net/man/1/xdg-email
If I'm reading this correct, "--subject" should be also used next to already used "--attachment" argument.

If you decide to implement it, would be nice if you could use file name **without** extension part. :)

Thank you for considering.

**Comment by DaddyAG** [comment]
Which part troubles you, file name part or "filename without extension" part?

In Windows, when you select one file (for example spreadsheet), right click on it and choose "send as attachment" you get subject field filled with whole file name "some spreadsheet.ods". Most of the time I choose to remove extension part so subject field is cleaner.

If you choose multiple files, then subject field is filled with all filenames separated with comma.

**Comment by rcalixte** [comment]
> Most of the time I choose to remove extension part so subject field is cleaner.

I understand that and it is a personal preference. Changing this option to suit your purposes changes it for everyone, but does _everyone_ want this? In general, I don't like the idea of making changes for the many to fit the needs of the one. This is also making an assumption that every period in a file name precedes a file extension. This is simply not true in the Linux world, especially since file extensions are optional in the first place.

> In Windows

Honestly, this part troubles me in addition to the previous issues. The goal here is not to imitate some functionality from Windows or assume that some level of prior art is the target.



In all, I'm trying to keep an open mind and see how to implement this so that everyone's preferences can be honored but without making this needlessly too complex to implement or maintain.

One option is that you could create a local copy on your machine and make your modifications there. That way, you have a personalized version that fits your precise needs. Does that at least work for you in the short-term?

**Comment by DaddyAG** [comment]
Yes, removing extension part is my personal preference. I don't know how others use it or what do they prefer.
As I mentioned, default with most windows programs is including everything (filename and extension).

If you see this base request as something beneficial to others it would be great if it would be implemented in any way or form. I don't mind removing .pdf or .ods/xls from email subjects, that was just "next step" in action evolution.

Maybe good broad use case would be:
1. If one file is picked, use full filename as subject.
2. If multiple files are picked, leave subject empty.

**Comment by DaddyAG** [comment]
This is what I'm using at home if you decide to change script. It takes filename from first file without extension and sets it as subject. Works for me. :)

`#!/usr/bin/env python3

import subprocess
import sys
import os

def main():
    '''
    Uzima listu argumenata i gradi xdg-email naredbu.
    '''
    if len(sys.argv) < 2:
        sys.exit(2)

    xdg_cmd = ['/usr/bin/xdg-email']

    # Uzmi ime prve datoteke bez ekstenzije kao predmet emaila
    first_file = sys.argv[1]
    subject = os.path.splitext(os.path.basename(first_file))[0]  # Skida ekstenziju

    # Dodaj predmet (naslov) emaila
    xdg_cmd.extend(['--subject', subject])

    # Dodaj sve datoteke kao privitke
    for file_arg in sys.argv[1:]:
        xdg_cmd.extend(['--attach', file_arg])
    
    try:
        subprocess.run(xdg_cmd, check=True)
    except subprocess.CalledProcessError as e:
        print(f"Došlo je do pogreške: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == '__main__':
    main()
`
